src/main.cpp

```diff
@@ -4559,10 +4616,11 @@ void UnloadBlockIndex()
     fHavePruned = false;
 }
 
-bool LoadBlockIndex()
+bool LoadBlockIndex(bool* fRebuildRequired)
 {
+    assert(fRebuildRequired != NULL);
     // Load block index from databases
-    if (!fReindex && !LoadBlockIndexDB())
+    if (!fReindex && !LoadBlockIndexDB(fRebuildRequired))
```

Comment:
maybe simplify to

    return fReindex || LoadBlockIndexDB(fRebuildRequired));